lms/djangoapps/teams/static/teams/js/views/teams_tab.js

```diff
@@ -7,304 +7,380 @@
-            'teams/js/views/topics',
             'teams/js/models/topic',
             'teams/js/collections/topic',
-            'teams/js/views/teams',
+            'teams/js/models/team',
             'teams/js/collections/team',
+            'teams/js/views/topics',
+            'teams/js/views/team_profile',
+            'teams/js/views/teams',
             'teams/js/views/edit_team',
             'text!teams/templates/teams_tab.underscore'],
-           function (Backbone, _, gettext, HeaderView, HeaderModel, TabbedView,
-                     TopicsView, TopicModel, TopicCollection, TeamsView, TeamCollection,
-                     TeamEditView, teamsTemplate) {
-               var ViewWithHeader = Backbone.View.extend({
-                   initialize: function (options) {
-                       this.header = options.header;
-                       this.main = options.main;
-                   },
+        function (Backbone, _, gettext, HeaderView, HeaderModel, TabbedView,
+                  TopicModel, TopicCollection, TeamModel, TeamCollection,
+                  TopicsView, TeamProfileView, TeamsView, TeamEditView,
+                  teamsTemplate) {
+            var ViewWithHeader = Backbone.View.extend({
+                initialize: function (options) {
+                    this.header = options.header;
+                    this.main = options.main;
+                },
+
+                render: function () {
+                    this.$el.html(_.template(teamsTemplate));
+                    this.$('p.error').hide();
+                    this.header.setElement(this.$('.teams-header')).render();
+                    this.main.setElement(this.$('.page-content')).render();
+                    return this;
+                }
+            });
+
+            var TeamTabView = Backbone.View.extend({
+                initialize: function(options) {
+                    var TempTabView, router;
+                    this.courseID = options.courseID;
+                    this.topics = options.topics;
+                    this.topicUrl = options.topicUrl;
+                    this.teamsUrl = options.teamsUrl;
+                    this.maxTeamSize = options.maxTeamSize;
+                    this.languages = options.languages;
+                    this.countries = options.countries;
+                    // This slightly tedious approach is necessary
+                    // to use regular expressions within Backbone
+                    // routes, allowing us to capture which tab
+                    // name is being routed to
+                    router = this.router = new Backbone.Router();
+                    _.each([
+                        [':default', _.bind(this.routeNotFound, this)],
+                        ['topics/:topic_id(/)', _.bind(this.browseTopic, this)],
+                        ['topics/:topic_id/create-team(/)', _.bind(this.newTeam, this)],
+                        ['teams/:topic_id/:team_id', _.bind(this.browseTeam, this)],
```

Comment:
Hmm, that must be new since I started this work. I've added the slash handling to match.